crates/storage/db-models/Cargo.toml

```diff
@@ -13,7 +13,7 @@ workspace = true
 
 [dependencies]
 # reth
-reth-codecs.workspace = true
+reth-codecs = { workspace = true, optional = true }
 reth-primitives-traits = { workspace = true, features = ["serde", "reth-codec"] }
```

Comment:
here we can likely also relax the "reth-codec" feature and only enable this if reth-codec is also enabled